### PR TITLE
Added support for the `PRPDC=` prefix to v6 branch

### PR DESCRIPTION
**Pull request by tdulcet**
Same as #246, but for your v6 branch. This is needed for @Danc2050 and I's PrimeNet script, which also supports GpuOwl v6.11 because it is allegedly faster for some users and still supports LL DC and standalone P-1. I got a request explicitly for this from user [techn1ciaN](https://www.mersenneforum.org/member.php?u=16918) on the Mersenne Forum.

* Added support for the `PRPDC=` prefix for PRP DC assignments, for consistency with Prime95/MPrime and Mlucas.
* Improved parsing of the assignment ID by separating the `N/A` case.
* Added support for PRP assignments without an assignment ID, for consistency with Prime95/MPrime and Mlucas.
* Added support for doing P-1 before LL DC assignments.
* Enabled `-O3` optimization in the Makefile.
* Copied the Continuous Integration (CI) from the master branch (#217) so these and any future changes would be tested.

See #246 and #217 for the full details. Note that this does build with Clang on Linux, as it precedes the regressing commit https://github.com/preda/gpuowl/commit/817a9831dc9fdc595709bd5f68482e654b758b65.

### CODE REVIEW

File: Worktodo.cpp [hunk 1]
```diff
@@ -98,12 +102,12 @@ std::optional<Task> Worktodo::getTask(Args &args) {
...
   // Try to get a task from the local worktodo.txt
   if (optional<Task> task = firstGoodTask(worktodoTxt)) {
-    if (task->kind == Task::PRP && task->wantsPm1) {
+    if ((task->kind == Task::LL and !task->wantsPm1) or (task->kind == Task::PRP and task->wantsPm1)) {
       // Some worktodo tasks can be expanded into subtasks:
       // PRP with wantsPm1>0 is expanded into a sequence of P-1 followed by PRP with wantsPm1==0.
       Task pm1{Task::PM1, task->exponent, task->AID, "", task->B1, task->B2, task->bitLo, task->wantsPm1};
       pm1.adjustBounds(args);
-      task->wantsPm1 = 0;
+      task->wantsPm1 = task->kind == Task::LL ? 1 : 0;
```
Comment (preda) [inline Worktodo.cpp:110]: I would paranthesize the condition. Maybe even the whole RHS expression.

**Review by preda — APPROVED** [review]

File: Worktodo.cpp [hunk 2]
```diff
@@ -98,12 +102,12 @@ std::optional<Task> Worktodo::getTask(Args &args) {
  again:
   // Try to get a task from the local worktodo.txt
   if (optional<Task> task = firstGoodTask(worktodoTxt)) {
-    if (task->kind == Task::PRP && task->wantsPm1) {
+    if ((task->kind == Task::LL and !task->wantsPm1) or (task->kind == Task::PRP and task->wantsPm1)) {
```
Comment (preda) [inline Worktodo.cpp:105]: I don't understand the logic here: if the task is LL and !wantsPm1, then you do P-1? and flip the "wants" flag to true in than case below?

File: Worktodo.cpp (hunk 1 above)
Comment (tdulcet) [reply Worktodo.cpp:110]: Do you mean you want this: `task->wantsPm1 = (task->kind == Task::LL ? 1 : 0);`?

This line could of course be simplified to just: `task->wantsPm1 = task->kind == Task::LL;`, but I thought that might be confusing.

File: Worktodo.cpp (hunk 2 above)
Comment (tdulcet) [reply Worktodo.cpp:105]: The `wantsPm1` value is read directly from the assignment line in the worktodo file. For PRP assignments, this is the `tests_saved` (PRP tests saved if factor is found), so a truthy (greater than zero) value means P-1 is needed. However, for LL assignments, this is the `has_been_pminus1ed` (has P-1 been done), so a falsy (zero) value means P-1 is needed, the opposite of PRP assignments. That is what this logic was trying to express, without changing your variable names, as I was trying to keep the changes to a minimum. Let me know if you would like me to make any changes...

File: Worktodo.cpp (hunk 1 above)
Comment (preda) [reply Worktodo.cpp:110]: No, I meant
wantsPm1 = (kind == LL) ? 1 : 0;
or even better:
wantsPm1 = (kind == LL);
Anyway not a big deal.

File: Worktodo.cpp (hunk 2 above)
Comment (preda) [reply Worktodo.cpp:105]: thanks for the explanation. The name "wantsPm1" ain't good, but I understand the motivation of keeping the change small.

**Comment by preda** [comment]
thanks!

**Pull request merged**